### PR TITLE
#704: Remove remaining ArtifactRepository leftovers + use version range for VersionsHelper where applicable

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Utilising the fact that we can pass the *versionRange* directly to `RepositorySystem.resolveVersionRange`, which should optimise the search process (possibly less lookups performed).

Also, removing the unused `findLatestVersion( Artifact, VersionRange,                                              Boolean, boolean )` method.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
I've noticed that there are still some `ArtifactRepository` leftovers left. Removed those.

Also, Maven 3 version range resolver allows version range resolving -- using that to speed up version resolution a bit.

Small refactoring/cleanup, especially in DisplayPluginUpdates.

@slawekjaranowski please check

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java [hunk 1]
```diff
@@ -194,19 +204,19 @@ public DisplayPluginUpdatesMojo( RepositorySystem repositorySystem,
      * @throws MojoExecutionException when things go wrong.
      */
     private Map<String, String> getSuperPomPluginManagement()
-        throws MojoExecutionException
+            throws MojoExecutionException
     {
-        if ( new DefaultArtifactVersion( "3.0" ).compareTo( runtimeInformation.getApplicationVersion() ) <= 0 )
+        if ( runtimeInformation.isMavenVersion( "[3,)" ) )
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java:209]: It always be true ... we require 3.2.5 as minimum

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java [hunk 2]
```diff
@@ -194,19 +204,19 @@ public DisplayPluginUpdatesMojo( RepositorySystem repositorySystem,
...
         {
             getLog().debug( "Using Maven 3.x strategy to determine superpom defined plugins" );
             try
             {
                 Method getPluginsBoundByDefaultToAllLifecycles =
-                    LifecycleExecutor.class.getMethod( "getPluginsBoundByDefaultToAllLifecycles",
-                            String.class );
+                        LifecycleExecutor.class.getMethod( "getPluginsBoundByDefaultToAllLifecycles",
+                                String.class );
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java:216]: Such reflections is probably not needed

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java [hunk 3]
```diff
@@ -408,11 +418,11 @@ public void execute()
...
 
         Set<Plugin> plugins = getProjectPlugins( superPomPluginManagement, parentPlugins, parentBuildPlugins,
-                                                 parentReportPlugins, pluginsWithVersionsSpecified );
+                parentReportPlugins, pluginsWithVersionsSpecified );
 
         List<String> pluginUpdates = new ArrayList<>();
         List<String> pluginLockdowns = new ArrayList<>();
-        ArtifactVersion curMavenVersion = runtimeInformation.getApplicationVersion();
+        ArtifactVersion curMavenVersion = new DefaultArtifactVersion( runtimeInformation.getMavenVersion() );
         ArtifactVersion specMavenVersion = MinimalMavenBuildVersionFinder.find( getProject(), "2.0", getLog() );
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java:426]: To investigate if code is needed for Maven 3.x

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java [hunk 4]
```diff
@@ -1111,76 +1069,58 @@ private Map<String, Plugin> getLifecyclePlugins( MavenProject project )
...
+     * @throws MojoFailureException thrown if plugin retrieval did not succeed
      */
     // pilfered this from enforcer-rules
     // TODO coordinate with Brian Fox to remove the duplicate code
     private Set<Plugin> getBoundPlugins( MavenProject project, String thePhases )
-        throws PluginNotFoundException, LifecycleExecutionException, IllegalAccessException
+            throws MojoExecutionException
     {
-        if ( new DefaultArtifactVersion( "3.0" ).compareTo( runtimeInformation.getApplicationVersion() ) <= 0 )
+        if ( runtimeInformation.isMavenVersion( "[3,)" ) )
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java:1079]: We use 3.2.5

**Comment by ajarmoniuk** [comment]
As advised in https://cwiki.apache.org/confluence/display/MAVEN/Plugin+migration+to+Maven3+dependencies

> Remove [maven-compat](https://maven.apache.org/ref/current/maven-compat/apidocs/) (or give it the test-scope if it is required by the maven-plugin-testing-harness)

**Comment by ajarmoniuk** [comment]
I guess I can move to PlexusJunit4TestCase in another PR.

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java (hunk 3 above)
Comment (ajarmoniuk) [reply versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java]: Changed from `2.0` to `3.2.5` as that is the minimum version. This is still necessary if the user is running a older than current version of Maven, but > 3.2.5

File: versions-maven-plugin/src/it/it-display-plugin-updates-001/pom.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
   <name>display-plugin-updates</name>
 
   <prerequisites>
-    <maven>2.0.6</maven>
+    <maven>3.2.5</maven>
   </prerequisites>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-display-plugin-updates-001/pom.xml:13]: prerequisites is only used for maven-plugin packaging.

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java [hunk 6]
```diff
@@ -442,7 +398,7 @@ public void execute()
             String effectiveVersion = version;
 
             Artifact artifactRange = getHelper().createPluginArtifact( plugin.getGroupId(), plugin.getArtifactId(),
-                                                                       version );
+                    version );
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/DisplayPluginUpdatesMojo.java:401]: I start thinking to introduce spotless as is done in Maven from parent poms 38 when we have such line changed 😄

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**